### ISSUE TITLE
Option to programmatically edit text

### ISSUE THREAD

**Issue by sparrow23** [kolkov/angular-editor#159]
I am trying to implement this editor. I want to replace the current selection and insert some text at the current cursor position.

**Comment by kolkov** [comment]
Hi! What should trigger the event? Click on the custom button or other event?

**Comment by sparrow23** [comment]
It should be a event which will be called, when user selected a word or line. This event should give index of the selected word or the index of the first word if line is selected. And when we are binding  htmlContent with ngModel, it should return only string. It is returning string with html tag.

**Comment by kolkov** [comment]
What do you mean by the word index in this context of discussion?

**Comment by sparrow23** [comment]
I meant, Let's say you selected a word in a line written in editor.  Now
that selected word you want to remove programatically. How we can do that.?
My approach is, if we will be having index of the word in a line. then we
can manipulate line based on the index programatically.

**Comment by kolkov** [comment]
Yes, everything would be very simple if there was simple text, but there are both tex and html tags there.

**Comment by sunilmkamble17** [comment]
Hi,

**I am trying to implement this editor. I want to replace the current selection and insert some text at the current cursor position.**

trying to implement the same functionality,
please let me know if you have any workaround.

**Comment by paddotk** [comment]
You can just edit the source text with javascript right? Like `myText.replace('replacable value', 'some other text')`

**Comment by Mary1717** [comment]
> o en la posi

hello, did you find any solution?
